source/shared/core_stmt.cpp

```diff
@@ -2553,9 +2554,15 @@ void finalize_output_parameters( _Inout_ sqlsrv_stmt* stmt TSRMLS_DC )
         }
         value_z = NULL;
     }  ZEND_HASH_FOREACH_END();
+    }
+    catch (core::CoreException&) {
+        // empty the hash table since it's been processed
+        zend_hash_clean(Z_ARRVAL(stmt->output_params));
 
+        throw;
```

Comment:
Should we DIE() here or just throw?

Reply:
No we should throw rather than die here because it might not be a fatal error at this point. Let the calling function(s) handle the exception(s)